process/message.go

```diff
@@ -192,12 +194,16 @@ func (prevote *Prevote) BlockHash() id.Hash {
 	return prevote.blockHash
 }
 
+func (prevote *Prevote) NilReasons() NilReasons {
+	return prevote.nilReasons
+}
+
 func (prevote *Prevote) Type() MessageType {
 	return PrevoteMessageType
 }
 
 func (prevote *Prevote) String() string {
-	return fmt.Sprintf("Prevote(Height=%v,Round=%v,BlockHash=%v)", prevote.Height(), prevote.Round(), prevote.BlockHash())
+	return fmt.Sprintf("Prevote(Height=%v,Round=%v,BlockHash=%v,NilReasons=%v)", prevote.Height(), prevote.Round(), prevote.BlockHash(), prevote.NilReasons())
```

Comment:
What does this string produce? `NilReasons` does not appear to implement `fmt.Stringer` so this will be producing some Go specific stringification of the map. It should be language agnostic, so preferrably the `NilReasons` are marshaled into binary, hashed, and stringified using non-padded base64. (See comment above about the need for deterministic iteration.)

Reply:
No need for it to be readable in that case?

Reply:
Well, ideally it is still readable but only in the sense that the `Prevote.String` method returns something that is readable overall. The content of `NilReasons=` would be nice to be readable, but that can left to logs.